lib/authorization.js

```diff
@@ -14,32 +14,35 @@ var authorizeWithRefreshSAS = function authorizeWithRefreshSAS(method, path, que
     debug("Refreshing shared-access-signature");
     // Avoid refreshing more than once
     this._nextSASRefresh = 0;
-    // Refresh SAS
-    this._sas = Promise.resolve(this.options.sas());
-    // Update _nextSASRefresh when the SAS has been refreshed
-    this._sas.then(function(sas) {
-      sas = querystring.parse(sas);
-      // Find next sas refresh time
-      self._nextSASRefresh = (
-        new Date(sas.se).getTime() - self.options.minSASAuthExpiry
-      );
-      debug("Refreshed shared-access-signature, will refresh in %s ms",
-        self._nextSASRefresh);
-      // Throw an error if the signature expiration comes too soon
-      if (Date.now() > self._nextSASRefresh) {
-        throw new Error("Refreshed SAS, but got a Shared-Access-Signature " +
-          "that expires less than options.minSASAuthExpiry " +
-          "from now, signature expiry: " + sas.se);
-      }
-    }).catch(function(err) {
-      // If we have an error freshing SAS that's bad and we'll emit it; for most
-      // apps it's probably best to ignore this error and just crash.
-      self.emit('error', err);
-    });
+    // Refresh SAS and update _nextSASRefresh when the SAS has been refreshed
+    this._sas = Promise.resolve(this.options.sas())
+      .then(function(sas) {
+        sas = querystring.parse(sas);
+        // Find next sas refresh time
+        self._nextSASRefresh = (
+            new Date(sas.se).getTime() - self.options.minSASAuthExpiry
+            );
+        debug("Refreshed shared-access-signature, will refresh at",
+            new Date(self._nextSASRefresh));
+        // Throw an error if the signature expiration comes too soon
+        if (Date.now() > self._nextSASRefresh) {
+          throw new Error("Refreshed SAS, but got a Shared-Access-Signature " +
+              "that expires less than options.minSASAuthExpiry (" +
+              self.options.minSASAuthExpiry / 1000 +
+              "s) from now; signature expiry: " + sas.se);
+        }
+      });
   }
 
+  // "fork" the `this._sas` promise
+  let rv = new Promise(function(resolve, reject) {
```

Comment:
What is this trying to do?

You know. It's okay to have two things waiting for the .then thing on a promise..

Reply:
Indeed, [you're right](http://www.ecma-international.org/ecma-262/6.0/#sec-performpromisethen).